chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/subscriptions/subjects/formatter/ServiceBaseUrlListChangedFormatter.java

```diff
@@ -0,0 +1,83 @@
+package gov.healthit.chpl.scheduler.job.subscriptions.subjects.formatter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import gov.healthit.chpl.activity.history.ListingActivityUtil;
+import gov.healthit.chpl.dao.ActivityDAO;
+import gov.healthit.chpl.domain.CertificationResult;
+import gov.healthit.chpl.domain.CertifiedProductSearchDetails;
+import gov.healthit.chpl.dto.ActivityDTO;
+import gov.healthit.chpl.subscription.domain.SubscriptionObservation;
+import gov.healthit.chpl.subscription.subject.processor.ServiceBaseUrlListChangedActivityProcessor;
+import gov.healthit.chpl.util.DateUtil;
+import gov.healthit.chpl.util.Util;
+import lombok.extern.log4j.Log4j2;
+
+@Component
+@Log4j2(topic = "subscriptionObservationsNotificationJobLogger")
+public class ServiceBaseUrlListChangedFormatter extends ObservationSubjectFormatter {
+    private static final String DESCRIPTION_REMOVED_UNFORMATTED = "Service Base URL List %s was removed from %s.";
+    private static final String DESCRIPTION_ADDED_UNFORMATTED = "Service Base URL List %s was added to %s.";
+    private static final String DESCRIPTION_UPDATED_UNFORMATTED = "Service Base URL List was changed from %s to %s on %s";
```

Comment:
What are the `_UNFORMATTED` tags for on these (and the other ones)? They seem redundant

Reply:
Just to indicate the message need to be properly formatted (with String.format, farther down in the class) to replace the %s values in the message with real text. I can change them